File: repositories/history_repository.go
package repositories

import (
	"context"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type VerificationHistoryRepository interface {
	Create(ctx context.Context, history *entities.VerificationHistory) error
	FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.VerificationHistory, error)
}

type GormVerificationHistoryRepository struct {
	db *gorm.DB
}

func NewVerificationHistoryRepository(db *gorm.DB) VerificationHistoryRepository {
	return &GormVerificationHistoryRepository{db: db}
}

func (r *GormVerificationHistoryRepository) Create(ctx context.Context, history *entities.VerificationHistory) error {
	return r.db.WithContext(ctx).Create(history).Error
}

func (r *GormVerificationHistoryRepository) FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.VerificationHistory, error) {
	var histories []entities.VerificationHistory
	err := r.db.WithContext(ctx).Where("doctor_id = ?", doctorID).Order("performed_at DESC").Find(&histories).Error
	return histories, err
}

File: backend_api/repositories/history_repository.py
from typing import List
from uuid import UUID
from sqlalchemy.orm import Session
from entities.history import VerificationHistory

class VerificationHistoryRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, history: VerificationHistory) -> None:
        self.db.add(history)
        self.db.commit()
        self.db.refresh(history)

    def FindByDoctorID(self, doctor_id: UUID) -> List[VerificationHistory]:
        return self.db.query(VerificationHistory).filter(
            VerificationHistory.doctor_id == str(doctor_id)
        ).order_by(VerificationHistory.performed_at.desc()).all()
